data/build.js

```diff
@@ -74,6 +74,73 @@ function createTreeSorter(name) {
   };
 }
 
+async function buildScientificNameToImageMap(names) {
+  function partition(arr, size) {
+    const numGroups = Math.ceil(arr.length / size);
+    const groups = new Array(numGroups).fill(null);
+
+    return groups.map((_, i) => arr.slice(i * size, i * size + size));
+  }
+
+  function buildRequestURL(titles) {
+    const BASE_URL = 'https://en.wikipedia.org/w/api.php';
+    const requestURL = new URL(BASE_URL);
+    requestURL.search = new URLSearchParams({
+      action: 'query',
+      format: 'json',
+      prop: 'pageimages',
+      piprop: 'thumbnail',
+      pithumbsize: '600',
+      titles,
+    });
+
+    return requestURL;
+  }
+
+  function getRequestURLs(data) {
+    const MAX_TITLES_PER_QUERY = 50;
+    const partitions = partition(data, MAX_TITLES_PER_QUERY);
+    const queryTitles = partitions.map((titles) => titles.join('|'));
+    const queries = queryTitles.map((titles) => buildRequestURL(titles));
+    return queries.map((query) => query.href);
+  }
+
+  async function getResponseJSON(url) {
+    const response = await fetch(url);
+    return response.json();
+  }
+
+  async function getImages(requestURLs) {
+    const images = {};
+    const responsesMap = requestURLs.map((url) => getResponseJSON(url));
+    const responses = await Promise.all(responsesMap);
+    const allPages = responses.reduce((accumulator, response) => {
+      const { query: { pages = [] } = {} } = response;
+      return accumulator.concat(...Object.values(pages));
+    }, []);
```

Comment:
We get the images of at most 50 trees per API call. Wait until all the APIs calls have been completed. Then combine the data received into one large array to parse all at once.

Reply:
I don't remember why I used concat here. Might have been part of the old linting rules? I'll research if there's a ESLint rule for concat vs. spread.